### PR TITLE
Add s2geography

### PR DESCRIPTION
**Pull request by benbovy**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- ~~If static libraries are linked in, the license of the static library is packaged.~~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by benbovy** [comment]
@paleolimbot do you want to join me as a maintainer of this conda recipe? (no worries at all if you prefer not)

**Comment by benbovy** [comment]
> I'll pass on the maintainer title for now (I would be useless for anything conda related anyway) but thanks for the invite!

No worries!

@conda-forge/help-c-cpp this is ready.

**Comment by jorisvandenbossche** [comment]
@benbovy I am happy to be added as maintainer

**Comment by benbovy** [comment]
@jorisvandenbossche great!

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
cc @ocefpaf in case you have some bandwidth

File: recipes/s2geography/bld.bat [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+                -DBUILD_SHARED_LIBS=ON ^
+                -DS2GEOGRAPHY_BUILD_EXAMPLES=OFF ^
+                -DS2GEOGRAPHY_BUILD_TESTS=OFF ^
+                -DS2GEOGRAPHY_CODE_COVERAGE=OFF ^
+                -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE
+if errorlevel 1 exit 1
+
+nmake
+if errorlevel 1 exit 1
+
```
Comment (ocefpaf) [inline recipes/s2geography/bld.bat:19]: Do we have a `make check` or `ctest` that we can run here?

**Comment by ocefpaf** [comment]
One minor question that can be addressed in the feedstock. Merging this.

**Comment by jorisvandenbossche** [comment]
Thanks a lot Filipe!

**Comment by benbovy** [comment]
Thanks @ocefpaf!

We'll enable running the tests for the Windows build on the feedstock repo. Currently test coverage is low, more tests have been added in downstream bindings libraries (a PR for the Python bindings will follow soon here).

**Pull request merged**